src/renderers/dom/shared/DOMProperty.js

```diff
@@ -197,33 +194,58 @@ var DOMProperty = {
   properties: {},
 
   /**
-   * Mapping from lowercase property names to the properly cased version, used
-   * to warn in the case of missing properties. Available only in __DEV__.
-   *
-   * autofocus is predefined, because adding it to the property whitelist
-   * causes unintended side effects.
-   *
-   * @type {Object}
+   * Checks whether a property name is a writeable attribute.
+   * @method
    */
-  getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
+  shouldSetAttribute: function(name, value) {
+    if (DOMProperty.isReservedProp(name)) {
+      return false;
+    }
 
-  /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
+    if (value === null) {
+      return true;
+    }
 
-  /**
-   * Checks whether a property name is a custom attribute.
-   * @method
-   */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+    var lowerCased = name.toLowerCase();
+
+    var propertyInfo = DOMProperty.properties[name];
+
+    switch (typeof value) {
+      case 'boolean':
+        if (propertyInfo) {
+          return true;
```

Comment:
@sebmarkbage I'd like to figure out how much `HAS_BOOLEAN_VALUE`, `HAS_NUMERIC_VALUE`, and `HAS_POSITIVE_NUMERIC_VALUE` are developer convenience vs ensuring correctness. I still don't understand why all of the fields we have to include to allow booleans aren't flagged with `HAS_BOOLEAN_VALUE`. 

Likewise for numeric fields (which we've dropped from the whitelist, except for things like `cols`, `rows`, `start`, etc. 

I _think_ this might have mattered back when they were assigned as properties. I'd have to do some archeology. 

Dan and I made the call to keep their property info the same as 15.x, but I still can't help but wonder.